pkg/web/apps/apps_psql_repository.go

```diff
@@ -19,16 +22,59 @@ func NewPostgreSQLRepository(db *sql.DB) Repository {
 
 // GetApps retrieves all apps from the database
 func (a *appsPostgreSQLRepository) GetApps() (*[]models.App, error) {
-	app1 := models.App{
-		ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
-		AppID:                 "com.aerogear.app1",
-		AppName:               "app1",
-		NumOfDeployedVersions: 1,
-		NumOfClients:          1,
-		NumOfAppLaunches:      1,
+	rows, err := a.db.Query(`SELECT id,app_id,app_name FROM app WHERE deleted_at IS NULL;`)
+
+	if err != nil {
+		log.Error(err)
+		return nil, models.ErrInternalServerError
 	}
 
-	apps := []models.App{app1}
+	defer func() {
+		err := rows.Close()
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	apps := []models.App{}
+	for rows.Next() {
+		var a models.App
+		if err = rows.Scan(&a.ID, &a.AppID, &a.AppName); err != nil {
+			log.Error(err)
+		}
+
+		apps = append(apps, a)
+	}
 
 	return &apps, nil
 }
+
+func (a *appsPostgreSQLRepository) GetAppVersions(id string) (*[]models.Version, error) {
+	rows, err := a.db.Query(`SELECT id,version,app_id,disabled,disabled_message,num_of_clients,num_of_app_launches FROM version WHERE app_id = $1`, id)
+
+	if err != nil {
+		log.Error(err)
+		return nil, models.ErrInternalServerError
+	}
+
+	defer func() {
+		err := rows.Close()
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	versions := []models.Version{}
+
+	// iterate over the rows and add the data to the array of versions
+	for rows.Next() {
```

Comment:
Has too many ways to solve the problems. In this imp you will spend too much memory to bring all data and to do the interactions, etc. It is possible to replace all in order to get the data required with just one select/query. You can do something like: 

```sql
SELECT
  tmp.app_pk as ID
  tmp.app_id as AppID,
  tmp.app_name as name,
  (count(*) group by tmp.app_pk,tmp.app_version_id) as NumOfDeployedVersions,
  (count(*) group by tmp.app_pk,tmp.device_id) as NumOfClients,
  (sum(tmp.launches) group by tmp.app_pk,tmp.app_version_id) as NumOfAppLaunches
FROM (
  SELECT
     a.id as app_pk,
     a.app_id as app_id,
     a.app_name as app_name,
     v.version_id as app_version_id,
     v.num_app_launches as launches,
     v.is_disabled as is_version_disabled,
     d.device_id as device_id,
     d.version_id as device_version_id
    FROM
      app as a,
      version as v,
      device as d
    WHERE
      AND a.deleted_at IS NOT NULL,
      AND a.app_id = v.app_id,
      AND v.version_id = d.version_id
) tmp
````
Also, if you find some issue doing as the above example then I'd like to suggest see how to use [INNER JOINS](https://www.postgresql.org/docs/8.3/tutorial-join.html).